src/Cms/Language.php

```diff
@@ -603,19 +642,36 @@ public function update(array $props = null): static
 		if ($updated->isDefault() === true) {
 			$kirby->defaultLanguage()?->clone(['default' => false])->save();
 
-			$code = $this->code();
 			$site = $kirby->site();
 
-			touch($site->contentFile($code));
+			if ($site->storage()->exists(VersionIdentifier::changes(), $this) === true) {
+				$site->storage()->touch(VersionIdentifier::changes(), $this);
+			}
+			if ($site->storage()->exists(VersionIdentifier::published(), $this) === true) {
+				$site->storage()->touch(VersionIdentifier::published(), $this);
+			}
 
 			foreach ($kirby->site()->index(true) as $page) {
 				$files = $page->files();
 
 				foreach ($files as $file) {
-					touch($file->contentFile($code));
+					if ($file->storage()->exists(VersionIdentifier::changes(), $this) === true) {
+						$file->storage()->touch(VersionIdentifier::changes(), $this);
+					}
+					if ($file->storage()->exists(VersionIdentifier::published(), $this) === true) {
+						$file->storage()->touch(VersionIdentifier::published(), $this);
+					}
 				}
 
-				touch($page->contentFile($code));
+				if ($page->storage()->exists(VersionIdentifier::changes(), $this) === true) {
+					$page->storage()->touch(VersionIdentifier::changes(), $this);
+				}
+				if (
+					$page->isDraft() === false &&
+					$page->storage()->exists(VersionIdentifier::published(), $this) === true
+				) {
+					$page->storage()->touch(VersionIdentifier::published(), $this);
+				}
```

Comment:
So far one of my design goals of the `ContentStorage` interface was to make it as strict as possible to make it as easy as possible to catch bugs from conflicting logic. I suggest we keep it like this for now. Once everything has settled, we can reduce the strictness again where it makes sense.

But TBH I'm not sure if it's a good idea to silently "not do anything" in an error case. Dropping too many exceptions or silently correcting method input has bitten us quite often in the past.

Reply:
While I agreed with you on the `ContentStorage::touch` probably throwing an exception, I just can't look passed how verbose this code is. It also makes the code hard to read as the nuances are easily missed between the cruft. Wondering if there is an easier way, that's still not unsafe in this specific case (where we don't care what happens if the file doesn't exist) that doesn't create such bloat.

Reply:
But if we need to merge before finding a solution because we're running out of time, can't hold this back. I'm quite unhappy that now time doesn't leave us with a chance to get these things right and I am afraid we won't get around to fix them later on.

Reply:
I agree the code is quite verbose at the moment because it directly calls the low-level storage methods. Maybe we can build mid-level methods to make this more DRY.

I've added a new section "clean-up tasks for later" to the concept and added a task there. If we collect such issues there, we will not forget about them.

Reply:
I’m doubtful we’ll get around to it later (resource wise) if we don’t do it right from the start. But what can I do with that time pressure…